### PR TITLE
Add PIT for audio metrics

### LINKED ISSUE #294: Add multi-speaker support to the audio metrics
## 🚀 Feature
Permutation invariant metrics

### Motivation

Multi-speaker tasks like speech separation need to find the best permutation of the speakers in target and the speakers in prediction, and also need to compute metrics in a permutation invariant way.

### Pitch

Implement the widely used Permutation Invariant Training (PIT) method

### PR DESCRIPTION
**Pull request by quancs**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

## What does this PR do?
Fixes #294.
Calculate audio metrics in a permutation invariant way in multi-speaker cases

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source/references/functional.rst [hunk 1]
```diff
@@ -9,6 +9,13 @@ Functional metrics
 Audio Metrics
 *************
 
+pit [func]
+~~~~~~~~~~~~~
```
Comment (Borda) [inline docs/source/references/functional.rst:13]:
```suggestion
~~~~~~~~~~
```

File: tests/audio/test_pit.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+from tests.helpers import seed_all
+from tests.helpers.testers import BATCH_SIZE, NUM_BATCHES, MetricTester
+from torchmetrics.audio import PIT
+from torchmetrics.functional import pit, si_sdr, snr
+from torchmetrics.utilities.imports import _TORCH_GREATER_EQUAL_1_6
+
+seed_all(42)
+
+Time = 10
```
Comment (Borda) [inline tests/audio/test_pit.py:32]:
```suggestion
TIME = 10
```
What kind of time is it, come delay, constact?

File: tests/audio/test_pit.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    preds=torch.rand(NUM_BATCHES, BATCH_SIZE, 3, Time),
+    target=torch.rand(NUM_BATCHES, BATCH_SIZE, 3, Time),
+)
+inputs2 = Input(
+    preds=torch.rand(NUM_BATCHES, BATCH_SIZE, 2, Time),
+    target=torch.rand(NUM_BATCHES, BATCH_SIZE, 2, Time),
+)
+
+
+def scipy_version(preds: Tensor, target: Tensor, metric_func: Callable, eval_func: str):
```
Comment (Borda) [inline tests/audio/test_pit.py:46]: docstring what does the func do? what is the return type?
by the name, I expect return scipy version but it does not seem so...

File: tests/audio/test_pit.py [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+
+
+def average_metric(preds: Tensor, target: Tensor, metric_func: Callable):
+    # shape: preds [BATCH_SIZE, 1, Time] , target [BATCH_SIZE, 1, Time]
+    # or shape: preds [NUM_BATCHES*BATCH_SIZE, 1, Time] , target [NUM_BATCHES*BATCH_SIZE, 1, Time]
+    return metric_func(preds, target)[0].mean()
+
+
+snr_pit_scipy = partial(scipy_version, metric_func=snr, eval_func='max')
+si_sdr_pit_scipy = partial(scipy_version, metric_func=si_sdr, eval_func='max')
```
Comment (Borda) [inline tests/audio/test_pit.py:71]:
```suggestion
scipy_snr_pit = partial(scipy_version, metric_func=snr, eval_func='max')
scipy_si_sdr_pit = partial(scipy_version, metric_func=si_sdr, eval_func='max')
```
lest be consistent with other `sklearn`s

File: torchmetrics/audio/pit.py [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+# limitations under the License.
+from typing import Any, Callable, Dict, Optional
+
+from torch import Tensor, tensor
+
+from torchmetrics.functional.audio.pit import pit
+from torchmetrics.metric import Metric
+
+
+class PIT(Metric):
```
Comment (Borda) [inline torchmetrics/audio/pit.py:22]: is it only for audio or rather a wrapper for other metrics, see
https://github.com/PyTorchLightning/metrics/blob/591bb2d7f92985efb321b54f96b2628c79e56fe5/torchmetrics/wrappers/bootstrapping.py#L49

File: torchmetrics/functional/audio/pit.py [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+
+import torch
+from torch import Tensor
+
+from torchmetrics.utilities.checks import _check_same_shape
+
+_ps_dict: dict = {}  # cache
+_ps_idx_dict: dict = {}  # cache
+
+_has_scipy = find_spec("scipy")
```
Comment (Borda) [inline torchmetrics/functional/audio/pit.py:26]: it is already here
https://github.com/PyTorchLightning/metrics/blob/591bb2d7f92985efb321b54f96b2628c79e56fe5/torchmetrics/utilities/imports.py#L78

File: torchmetrics/functional/audio/pit.py [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+from itertools import permutations
+from typing import Any, Callable, Dict, Tuple, Union
+
+import torch
+from torch import Tensor
+
+from torchmetrics.utilities.checks import _check_same_shape
+
+_ps_dict: dict = {}  # cache
+_ps_idx_dict: dict = {}  # cache
```
Comment (Borda) [inline torchmetrics/functional/audio/pit.py:24]: some more info what kind of cache?

File: torchmetrics/functional/audio/pit.py [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+    from scipy.optimize import linear_sum_assignment
+    mmtx = metric_mtx.detach().cpu()
+    best_perm = torch.tensor([linear_sum_assignment(pwm, eval_func == torch.max)[1] for pwm in mmtx])
+    best_perm = best_perm.to(metric_mtx.device)
+    best_metric = torch.gather(metric_mtx, 2, best_perm[:, :, None]).mean([-1, -2])
+    return best_metric, best_perm  # shape [batch], shape [batch, spk]
+
+
+def _find_best_perm_by_exhuastive_method(metric_mtx: torch.Tensor,
+                                         eval_func: Union[torch.min, torch.max]) -> Tuple[Tensor, Tensor]:
```
Comment (Borda) [inline torchmetrics/functional/audio/pit.py:42]:
```suggestion
                                         eval_func: Union[torch.min, torch.max],) -> Tuple[Tensor, Tensor]:
```

File: torchmetrics/functional/audio/pit.py [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+) -> Tuple[Tensor, Tensor]:
+    from scipy.optimize import linear_sum_assignment
+    mmtx = metric_mtx.detach().cpu()
+    best_perm = torch.tensor([linear_sum_assignment(pwm, eval_func == torch.max)[1] for pwm in mmtx])
+    best_perm = best_perm.to(metric_mtx.device)
+    best_metric = torch.gather(metric_mtx, 2, best_perm[:, :, None]).mean([-1, -2])
+    return best_metric, best_perm  # shape [batch], shape [batch, spk]
+
+
+def _find_best_perm_by_exhuastive_method(metric_mtx: torch.Tensor,
```
Comment (Borda) [inline torchmetrics/functional/audio/pit.py:41]: add docstring with descrition

File: tests/audio/test_pit.py (hunk 2 above)
Comment (quancs) [reply tests/audio/test_pit.py:32]: The time length of signal

File: torchmetrics/functional/audio/pit.py (hunk 7 above)
Comment (quancs) [reply torchmetrics/functional/audio/pit.py]: edited see https://github.com/PyTorchLightning/metrics/pull/384/commits/105957524666a2e73fc1bbdb7beaa9e4de67e7c4

Comment (quancs) [reply torchmetrics/functional/audio/pit.py]: that is the simplified version

File: torchmetrics/functional/audio/pit.py (hunk 6 above)
Comment (quancs) [reply torchmetrics/functional/audio/pit.py]: see use _SCIPY_AVAILABLE: https://github.com/PyTorchLightning/metrics/pull/384/commits/16d70f9385253fdf89f4a0acb0432105dc0080d6

File: tests/audio/test_pit.py (hunk 2 above)
Comment (quancs) [reply tests/audio/test_pit.py]: see https://github.com/PyTorchLightning/metrics/pull/384/commits/6367adc28b53b511cfd63b9958fd7156854adbf0

File: tests/audio/test_pit.py (hunk 3 above)
Comment (quancs) [reply tests/audio/test_pit.py]: see https://github.com/PyTorchLightning/metrics/pull/384/commits/29744ab59371fdef6fd1e3b7b803b2a72b6078a8

File: torchmetrics/functional/audio/pit.py (hunk 9 above)
Comment (quancs) [reply torchmetrics/functional/audio/pit.py]: see https://github.com/PyTorchLightning/metrics/pull/384/commits/b452792349e84d5b4e3d1f2404fcc2bc5a35a64e

File: tests/audio/test_pit.py (hunk 4 above)
Comment (quancs) [reply tests/audio/test_pit.py]: the name `scipy_version ` changed to `a_naive_implementation_of_pit_based_on_scipy` in https://github.com/PyTorchLightning/metrics/pull/384/commits/29744ab59371fdef6fd1e3b7b803b2a72b6078a8. maybe here we can usse `snr_naive_pit` and `si_sdr_naive_pit`?

File: torchmetrics/audio/pit.py (hunk 5 above)
Comment (quancs) [reply torchmetrics/audio/pit.py:22]: It is only for audio. The PIT implements the famous Permutation Invariant Training method in speech separation field in order to calculate audio metrics in a permutation invariant way.

File: torchmetrics/functional/audio/pit.py (hunk 9 above)
Comment (quancs) [reply torchmetrics/functional/audio/pit.py]: also see https://github.com/PyTorchLightning/metrics/pull/384/commits/859e25e3218b8ca31b54a8f8a49bb95bb882117e

Comment (quancs) [reply torchmetrics/functional/audio/pit.py]: The last one is the final one ...

File: tests/audio/test_pit.py [hunk 10]
```diff
@@ -0,0 +1,210 @@
...
+)
+
+
+def a_naive_implementation_of_pit_based_on_scipy(
+    preds: Tensor,
+    target: Tensor,
+    metric_func: Callable,
+    eval_func: str,
+) -> Tuple[Tensor, Tensor]:
+    """a naive implementation of pit based on scipy
```
Comment (SkafteNicki) [inline tests/audio/test_pit.py:54]: do you have a reference for this or did you just implement it yourself?

File: tests/audio/test_pit.py [hunk 11]
```diff
@@ -0,0 +1,210 @@
...
+            preds,
+            target,
+            PIT,
+            sk_metric=partial(average_metric, metric_func=sk_metric),
+            dist_sync_on_step=dist_sync_on_step,
+            metric_args=dict(metric_func=metric_func, eval_func=eval_func),
+        )
+
+    def test_pit_functional(self, preds, target, sk_metric, metric_func, eval_func):
+        device = 'cuda' if (torch.cuda.is_available() and torch.cuda.device_count() > 0) else 'cpu'
```
Comment (SkafteNicki) [inline tests/audio/test_pit.py:129]: any reason you cannot use the `self.run_functional_metric_test` method here?

File: torchmetrics/audio/pit.py (hunk 5 above)
Comment (SkafteNicki) [reply torchmetrics/audio/pit.py:22]: @quancs could you add what you just wrote to the docstring? seems great to establish what it is used for

File: torchmetrics/functional/audio/pit.py [hunk 12]
```diff
@@ -0,0 +1,209 @@
...
+                 [-0.1719,  0.3205,  0.2951]]])
+
+    Reference:
+        [1]	D. Yu, M. Kolbaek, Z.-H. Tan, J. Jensen, Permutation invariant training of deep models for
+        speaker-independent multi-talker speech separation, in: 2017 IEEE Int. Conf. Acoust. Speech
+        Signal Process. ICASSP, IEEE, New Orleans, LA, 2017: pp. 241–245. https://doi.org/10.1109/ICASSP.2017.7952154.
+    """
+    _check_same_shape(preds, target)
+    if eval_func not in ['max', 'min']:
+        raise RuntimeError('eval_func can only be "max" or "min"')
```
Comment (SkafteNicki) [inline torchmetrics/functional/audio/pit.py:152]: remember to update tests!
```suggestion
        raise ValueError(f'eval_func can only be "max" or "min" but got {eval_func}')
```

File: torchmetrics/functional/audio/pit.py [hunk 13]
```diff
@@ -0,0 +1,209 @@
...
+    if len(target.shape) < 2:
+        raise RuntimeError(f"Inputs must be of shape [batch, spk, ...], got {target.shape} and {preds.shape} instead")
```
Comment (SkafteNicki) [inline torchmetrics/functional/audio/pit.py:154]: remember to update tests!
```suggestion
        raise ValueError(f"Inputs must be of shape [batch, spk, ...], got {target.shape} and {preds.shape} instead")
```

File: torchmetrics/functional/audio/pit.py [hunk 14]
```diff
@@ -0,0 +1,209 @@
...
+    batch_size, spk_num = target.shape[0:2]
+    metric_mtx = torch.empty((batch_size, spk_num, spk_num), dtype=preds.dtype, device=target.device)
+    for t in range(spk_num):
+        for e in range(spk_num):
+            metric_mtx[:, t, e] = metric_func(preds[:, e, ...], target[:, t, ...], **kwargs)
+
+    # find best
+    if spk_num < 3 or _SCIPY_AVAILABLE is None:
+        if spk_num >= 3 and _SCIPY_AVAILABLE is None:
+            import warnings
```
Comment (SkafteNicki) [inline torchmetrics/functional/audio/pit.py:166]: move import to top of file
```suggestion
```

File: torchmetrics/functional/audio/pit.py [hunk 15]
```diff
@@ -0,0 +1,209 @@
...
+            warnings.warn(f"Speaker-num is {spk_num}>3, you'd better install scipy to improve pit's performance!")
```
Comment (SkafteNicki) [inline torchmetrics/functional/audio/pit.py:167]: just personal preference 
```suggestion
            warnings.warn(f"In pit metric for speaker-num {spk_num}>3, we recommend installing scipy for better performance")
```

File: torchmetrics/functional/audio/pit.py [hunk 16]
```diff
@@ -0,0 +1,209 @@
...
+    else:
+        best_metric, best_perm = _find_best_perm_by_linear_sum_assignment(
+            metric_mtx,
+            torch.max if eval_func == 'max' else torch.min,
+        )
+
+    return best_metric, best_perm
+
+
+def permutate(preds: Tensor, perm: Tensor) -> Tensor:
```
Comment (SkafteNicki) [inline torchmetrics/functional/audio/pit.py:182]: It seems this metric is public and users are expected to use it?
Then it should probably be included in the docs?
Also if this is the case we may want to call it something more specific like `pit_permutate`

File: tests/audio/test_pit.py (hunk 11 above)
Comment (quancs) [reply tests/audio/test_pit.py]: done. see https://github.com/PyTorchLightning/metrics/pull/384/commits/a4e347a34793d97c824ebfba922e304af949e27d

File: torchmetrics/audio/pit.py (hunk 5 above)
Comment (quancs) [reply torchmetrics/audio/pit.py:22]: done. see https://github.com/PyTorchLightning/metrics/pull/384/commits/01d484edabd99d35c988c6615845edb00d5bd40c

File: tests/audio/test_pit.py (hunk 10 above)
Comment (quancs) [reply tests/audio/test_pit.py:54]: It's just implemented by myself. The idea of PIT is very simple: find the best (minimal or maximal) metric among all possible permutations of preds and target. Thus, it is equivalent to calculate the metric matrix first, and then solve the linear sum assignment problem.

[pytorch-pit](https://github.com/asteroid-team/pytorch-pit) is the pit used in [asteroid](https://github.com/asteroid-team/asteroid).  Should I change to use it to verify my implementation?

Comment (SkafteNicki) [reply tests/audio/test_pit.py:54]: No, I am fine with this explanation that you just gave. Maybe you could add it to the docstring of the function?

File: docs/source/references/modules.rst [hunk 17]
```diff
@@ -40,6 +40,12 @@ the metric will be computed over the ``time`` dimension.
     >>> snr_val
     tensor(16.1805)
 
+PIT
+~~~~~~
```
Comment (Borda) [inline docs/source/references/modules.rst:44]:
```suggestion
~~~
```

File: tests/audio/test_pit.py [hunk 18]
```diff
@@ -0,0 +1,202 @@
...
+    target=torch.rand(NUM_BATCHES, BATCH_SIZE, 3, TIME),
+)
+# two speaker examples to test _find_best_perm_by_exhuastive_method
+inputs2 = Input(
+    preds=torch.rand(NUM_BATCHES, BATCH_SIZE, 2, TIME),
+    target=torch.rand(NUM_BATCHES, BATCH_SIZE, 2, TIME),
+)
+
+
+def a_naive_implementation_of_pit_based_on_scipy(
```
Comment (Borda) [inline tests/audio/test_pit.py:48]:
```suggestion
def naive_implementation_pit_scipy(
```

File: tests/audio/test_pit.py [hunk 19]
```diff
@@ -0,0 +1,202 @@
...
+        metric_func: a function which return best_metric and best_perm
+
+    Returns:
+        the average of best_metric
+    """
+    return metric_func(preds, target)[0].mean()
+
+
+snr_pit_scipy = partial(a_naive_implementation_of_pit_based_on_scipy, metric_func=snr, eval_func='max')
+si_sdr_pit_scipy = partial(a_naive_implementation_of_pit_based_on_scipy, metric_func=si_sdr, eval_func='max')
```
Comment (Borda) [inline tests/audio/test_pit.py:100]:
```suggestion
snr_pit_scipy = partial(naive_implementation_pit_scipy, metric_func=snr, eval_func='max')
si_sdr_pit_scipy = partial(naive_implementation_pit_scipy, metric_func=si_sdr, eval_func='max')
```

File: tests/audio/test_pit.py (hunk 10 above)
Comment (Borda) [reply tests/audio/test_pit.py:54]: could you pls add the same explanation to the function docstring?

File: tests/audio/test_pit.py [hunk 20]
```diff
@@ -0,0 +1,202 @@
...
+    best_metrics = []
+    best_perms = []
+    for b in range(batch_size):
+        row_idx, col_idx = linear_sum_assignment(metric_mtx[b, ...], eval_func == 'max')
+        best_metrics.append(metric_mtx[b, row_idx, col_idx].mean())
+        best_perms.append(col_idx)
+    return torch.from_numpy(np.stack(best_metrics)), torch.from_numpy(np.stack(best_perms))
+
+
+def average_metric(preds: Tensor, target: Tensor, metric_func: Callable) -> Tensor:
```
Comment (Borda) [inline tests/audio/test_pit.py:85]:
```suggestion
def _average_metric(preds: Tensor, target: Tensor, metric_func: Callable) -> Tensor:
```
rather protected

File: torchmetrics/audio/pit.py [hunk 21]
```diff
@@ -0,0 +1,115 @@
...
+        average PIT metric
+
+    Example:
+        >>> import torch
+        >>> from torchmetrics import PIT
+        >>> from torchmetrics.functional.audio import si_snr
+        >>> preds = torch.randn(3, 2, 5) # [batch, spk, time]
+        >>> target = torch.randn(3, 2, 5) # [batch, spk, time]
+        >>> pit = PIT(si_snr, 'max')
+        >>> avg_pit_metric = pit(preds, target)
```
Comment (Borda) [inline torchmetrics/audio/pit.py:61]:
```suggestion
        >>> pit(preds, target)
```
of what is wrong with the number?

File: torchmetrics/functional/audio/pit.py [hunk 22]
```diff
@@ -0,0 +1,212 @@
...
+        >>> # [batch, spk, time]
+        >>> preds = torch.tensor([[[-0.0579,  0.3560, -0.9604], [-0.1719,  0.3205,  0.2951]]])
+        >>> target = torch.tensor([[[ 1.0958, -0.1648,  0.5228], [-0.4100,  1.1942, -0.5103]]])
+        >>> best_metric, best_perm = pit(preds, target, si_sdr, 'max')
+        >>> best_metric
+        tensor([-5.1091])
+        >>> best_perm
+        tensor([[0, 1]])
+        >>> preds_pmted = permutate(preds, best_perm)
+        >>> preds_pmted
```
Comment (Borda) [inline torchmetrics/functional/audio/pit.py:143]:
```suggestion
        >>> pit_permutate(preds, best_perm)
```

File: torchmetrics/functional/audio/pit.py [hunk 23]
```diff
@@ -0,0 +1,212 @@
...
```
Comment (Borda) [inline torchmetrics/functional/audio/pit.py:207]:
```suggestion
        >>> pit_permutate(preds, best_perm)
```

File: torchmetrics/functional/audio/pit.py [hunk 24]
```diff
@@ -0,0 +1,212 @@
...
+        kwargs:
+            additional args for metric_func
+
+    Returns:
+        best_metric of shape [batch],
+        best_perm of shape [batch]
+
+    Example:
+        >>> import torch
+        >>> from torchmetrics.functional.audio import si_sdr, pit, permutate
```
Comment (Borda) [inline torchmetrics/functional/audio/pit.py:133]:
```suggestion
        >>> from torchmetrics.functional.audio import si_sdr
```

File: torchmetrics/functional/audio/pit.py [hunk 25]
```diff
@@ -0,0 +1,212 @@
...
+    Args:
+        preds (Tensor): the estimates you want to permutate, shape [batch, spk, ...]
+        perm (Tensor): the permutation returned from pit, shape [batch, spk]
+
+    Returns:
+        Tensor: the permutated version of estimate
+
+    Example:
+        >>> import torch
+        >>> from torchmetrics.functional.audio import si_sdr, pit, permutate
```
Comment (Borda) [inline torchmetrics/functional/audio/pit.py:197]:
```suggestion
        >>> from torchmetrics.functional.audio import si_sdr
```

**Review by Borda — APPROVED** [review]

File: tests/audio/test_pit.py [hunk 26]
```diff
@@ -0,0 +1,202 @@
...
+)
+
+
+def naive_implementation_pit_scipy(
+    preds: Tensor,
+    target: Tensor,
+    metric_func: Callable,
+    eval_func: str,
+) -> Tuple[Tensor, Tensor]:
+    """a naive implementation of pit based on scipy
```
Comment (tchaton) [inline tests/audio/test_pit.py:54]:
```suggestion
    """A naive implementation of `Permutation Invariant Training` based on Scipy
```

**Review by tchaton — APPROVED** [review]
LGTM !

File: torchmetrics/functional/audio/pit.py [hunk 27]
```diff
@@ -0,0 +1,203 @@
...
+
+    Reference:
+        [1]	D. Yu, M. Kolbaek, Z.-H. Tan, J. Jensen, Permutation invariant training of deep models for
+        speaker-independent multi-talker speech separation, in: 2017 IEEE Int. Conf. Acoust. Speech
+        Signal Process. ICASSP, IEEE, New Orleans, LA, 2017: pp. 241–245. https://doi.org/10.1109/ICASSP.2017.7952154.
+    """
+    _check_same_shape(preds, target)
+    if eval_func not in ['max', 'min']:
+        raise ValueError(f'eval_func can only be "max" or "min" but got {eval_func}')
+    if len(target.shape) < 2:
```
Comment (SkafteNicki) [inline torchmetrics/functional/audio/pit.py:153]:
```suggestion
    if target.ndim < 2:
```

File: torchmetrics/functional/audio/pit.py [hunk 28]
```diff
@@ -0,0 +1,203 @@
...
+
+def pit(
+    preds: torch.Tensor,
+    target: torch.Tensor,
+    metric_func: Callable,
+    eval_func: str = 'max',
+    **kwargs: Dict[str, Any]
+) -> Tuple[Tensor, Tensor]:
+    """ Permutation invariant training (PIT). The PIT implements the famous Permutation Invariant Training method in
+    speech separation field in order to calculate audio metrics in a permutation invariant way.
```
Comment (SkafteNicki) [inline torchmetrics/functional/audio/pit.py:111]:
```suggestion
    """ 
    Permutation invariant training (PIT). The PIT implements the famous Permutation Invariant Training method [1]
    in speech separation field in order to calculate audio metrics in a permutation invariant way.
```

File: torchmetrics/audio/pit.py [hunk 29]
```diff
@@ -0,0 +1,117 @@
...
+
+from torch import Tensor, tensor
+
+from torchmetrics.functional.audio.pit import pit
+from torchmetrics.metric import Metric
+
+
+class PIT(Metric):
+    """ Permutation invariant training (PIT). The PIT implements the famous Permutation Invariant Training method in
+    speech separation field in order to calculate audio metrics in a permutation invariant way.
```
Comment (SkafteNicki) [inline torchmetrics/audio/pit.py:24]:
```suggestion
    """ 
    Permutation invariant training (PIT). The PIT implements the famous Permutation Invariant Training method [1]
    in speech separation field in order to calculate audio metrics in a permutation invariant way.
```

**Review by SkafteNicki — APPROVED** [review]
LGTM

**Pull request merged**